rootfs/etc/nginx/lua/certificate.lua

```diff
@@ -18,30 +19,31 @@ local DEFAULT_CERT_HOSTNAME = "_"
 local certificate_data = ngx.shared.certificate_data
 local certificate_servers = ngx.shared.certificate_servers
 local ocsp_response_cache = ngx.shared.ocsp_response_cache
+local certificate_cache
 
-local function get_der_cert_and_priv_key(pem_cert_key)
-  local der_cert, der_cert_err = ssl.cert_pem_to_der(pem_cert_key)
-  if not der_cert then
-    return nil, nil, "failed to convert certificate chain from PEM to DER: " .. der_cert_err
+local function get_cert_and_priv_key(pem_cert_key)
+  local cert, cert_err = ssl.parse_pem_cert(pem_cert_key)
+  if not cert then
+    return nil, nil, "failed to parse PEM certificate chain: " .. cert_err
   end
 
-  local der_priv_key, dev_priv_key_err = ssl.priv_key_pem_to_der(pem_cert_key)
-  if not der_priv_key then
-    return nil, nil, "failed to convert private key from PEM to DER: " .. dev_priv_key_err
+  local priv_key, priv_key_err = ssl.parse_pem_priv_key(pem_cert_key)
+  if not priv_key then
+    return nil, nil, "failed to parse PEM private key: " .. priv_key_err
   end
 
-  return der_cert, der_priv_key, nil
+  return cert, priv_key, nil
 end
 
-local function set_der_cert_and_key(der_cert, der_priv_key)
-  local set_cert_ok, set_cert_err = ssl.set_der_cert(der_cert)
+local function set_cert_and_key(cert, priv_key)
+  local set_cert_ok, set_cert_err = ssl.set_cert(cert)
```

Comment:
> Note that this set_cert function will run slightly faster, in terms of CPU cycles wasted, than the [set_der_cert](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#set_der_cert) variant, since the first function uses opaque cdata pointers which do not require any additional conversion needed to be performed by the SSL library during the SSL handshake.